src/eigenvalue.F90

```diff
@@ -458,73 +458,131 @@ subroutine calculate_combined_keff()
+
+      ! Initialize variables
+      g = ZERO
+      S = ZERO
+      k_combined = ZERO
+
+      do l = 1, 3
+        ! Permutations of estimates
+        if (l == 1) then
+          ! i = collision, j = absorption, k = tracklength
+          i = 1
+          j = 2
+          k = 3
+        elseif (l == 2) then
+          ! i = absortion, j = tracklength, k = collision
+          i = 2
+          j = 3
+          k = 1
+        elseif (l == 3) then
+          ! i = tracklength, j = collision, k = absorption
+          i = 3
+          j = 1
+          k = 2
+        end if
 
-      ! Add to S sums for variance of combined estimate
-      S(1) = S(1) + f * cov(1,l)
-      S(2) = S(2) + (cov(j,j) + cov(k,k) - TWO*cov(j,k))*kv(l)*kv(l)
-      S(3) = S(3) + (cov(k,k) + cov(i,j) - cov(j,k) - cov(i,k))*kv(l)*kv(j)
+        ! Calculate weighting
+        f = cov(j, j) * (cov(k, k) - cov(i, k)) - cov(k, k) * cov(i, j) + &
+             cov(j, k) * (cov(i, j) + cov(i, k) - cov(j, k))
 
-      ! Add to sum for combined k-effective
-      k_combined(1) = k_combined(1) + f * kv(l)
-      g = g + f
-    end do
+        ! Add to S sums for variance of combined estimate
+        S(1) = S(1) + f * cov(1, l)
+        S(2) = S(2) + (cov(j, j) + cov(k, k) - TWO * cov(j, k)) * kv(l) * kv(l)
+        S(3) = S(3) + (cov(k, k) + cov(i, j) - cov(j, k) - &
+             cov(i, k)) * kv(l) * kv(j)
+
+        ! Add to sum for combined k-effective
+        k_combined(1) = k_combined(1) + f * kv(l)
+        g = g + f
+      end do
 
-    ! Complete calculations of S sums
-    S = (n - 1)*S
-    S(1) = (n - 1)**2 * S(1)
+      ! Complete calculations of S sums
+      S = (n - ONE) * S
+      S(1) = (n - ONE)**2 * S(1)
 
-    ! Calculate combined estimate of k-effective
-    k_combined(1) = k_combined(1) / g
+      ! Calculate combined estimate of k-effective
+      k_combined(1) = k_combined(1) / g
 
-    ! Calculate standard deviation of combined estimate
-    g = (n - 1)**2 * g
-    k_combined(2) = sqrt(S(1)/(g*n*(n-3)) * (ONE + n*((S(2) - TWO*S(3))/g)))
+      ! Calculate standard deviation of combined estimate
+      g = (n - ONE)**2 * g
+      k_combined(2) = sqrt(S(1) / &
+           (g * n * (n - THREE)) * (ONE + n * ((S(2) - TWO * S(3)) / g)))
+
+    else
+      ! Use only two estimators
+      ! These equations are derived analogously to that done in the paper by
+      ! Urbatsch, but are simpler than for the three estimators case since the
+      ! block matrices of the three estimator equations reduces to scalars here
+
+      ! Store commonly used terms
+      f = cov(i, i) - cov(i, j)
+      g = (cov(i, i) + cov(j, j) - TWO * cov(i, j))
+
+      ! Calculate combined estimate of k-effective
+      k_combined(1) = kv(i) - f / g * (kv(i) - kv(j))
+
+      ! Calculate standard deviation of combined estimate
+      k_combined(2) = sqrt(cov(i, i) * (ONE - f * f / g) * &
+           (ONE / n + (kv(i) - kv(j)) * (kv(i) - kv(j)) / ((n - ONE) * g)))
```

Comment:
I'm not able to get this to match your expression from the Jupyter notebook. More importantly, it doesn't contain N - 2 as I would expect. Check out Eq. 50 in [Urbatsch's technical report](https://laws.lanl.gov/vhosts/mcnp.lanl.gov/pdf_files/la-12658.pdf).